File: intersectionOfTwoArrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = [x for x in nums1 if x in nums2]
        return set(result)

# better solution

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        common = set(nums1)
        result = []
        for i in nums2:
            if i in common:
                result.append(i)
                common.remove(i)
        return result

File: intersectionOfTwoArrays.cpp
class Solution
{
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
    {
        unordered_set<int> set1, set2;
        for (auto i : nums1)
            set1.insert(i);
        for (auto i : nums2)
        {
            if (set1.find(i) != set1.end())
                set2.emplace(i);
        }
        return {set2.begin(), set2.end()};
    }
};